### DISCUSSION TITLE
IndexOptimize minimum fragmentation level = 0?

### DISCUSSION THREAD

**Discussion by Sporidium** [olahallengren/sql-server-maintenance-solution#342] [Questions & Answers]
Is there any reason why a value of zero is disallowed for the @FragmentationLevel1 parameter of the IndexOptimize procedure?
I know I could make a very simple tweak to the parameter validation code, but I'd prefer to use the tools "out of the box" without modifying them.
The problem arises with a large table to which new rows are added at the end of the clustered index, and old rows are deleted from the beginning.  I believe (I don't have direct access yet) this is resulting in very little fragmentation as measured by the index physical stats DMV, but is causing unused space to grow and grow to a point where an index reorganization is needed to reclaim that space.

**Comment by olahallengren** [comment]
I am putting the indexes into one of three fragmentation groups (low, medium and high). @FragmentationLevel1 is the threshold between low and medium fragmentation.

If you want to rebuild indexes with a low fragmentation, you can add this parameter:
```@FragmentationLow = 'INDEX_REBUILD_ONLINE,INDEX_REBUILD_OFFLINE'```

**Comment by Sporidium** [comment]
Understood.  Quite an elegant interface, pity I couldn't see how to use it!
Closing the issue.
